README.md

```diff
@@ -1,7 +1,139 @@
 # Tapable
 
+The tapable packages exposes many Hook classes, which can be used to create hooks for plugins.
+
 ``` javascript
-var Tapable = require("tapable");
+const {
+	SyncHook,
+	SyncBailHook,
+	SyncWaterfallHook,
+	SyncLoopHook,
+	AsyncParallelHook,
+	AsyncParallelBailHook,
+	AsyncSequencialHook,
+	AsyncSequencialBailHook,
+	AsyncWaterfallHook
+ } = require("tapable");
+```
+
+## Usage
+
+All Hook constructors take one optional argument, which is a list of argument names as strings.
+
+``` js
+const hook = new SyncHook(["arg1", "arg2", "arg3"]);
+```
+
+The best practice is to expose all hooks of a class in a `hooks` property:
+
+``` js
+class Car {
```

Comment:
Generic would be better here imho, the `Car` example could/should be under examples instead imho

```js
class Plugin { ... }
```